sunpy/image/transform.py

```diff
@@ -53,29 +53,8 @@ def affine_transform(image, rmatrix, order=3, scale=1.0, image_center=None,
 
     Notes
     -----
-    This algorithm uses an affine transformation as opposed to a polynomial
-    geometrical transformation, which by default is `skimage.transform.warp`.
-    One can specify using `scipy.ndimage.affine_transform` as
-    an alternative affine transformation. The two transformations use different
-    algorithms and thus do not give identical output.
-
-    When using `skimage.transform.warp` with order >= 4 or using
-    `scipy.ndimage.affine_transform` at all, "NaN" values will be replaced with
-    zero prior to rotation. No attempt is made to retain the "NaN" values.
-
-    Input arrays with integer data are cast to float 64 and can be re-cast using
-    `numpy.ndarray.astype` if desired.
-
-    In the case of `skimage.transform.warp`, the image is normalized to [0, 1]
-    before passing it to the function. It is later rescaled back to the original range.
-
-    Although this function is analogous to the IDL's ``rot`` function, it does not
-    use the same algorithm as the IDL ``rot`` function.
-    IDL's ``rot`` calls the `POLY_2D <https://www.harrisgeospatial.com/docs/poly_2d.html>`__
-    method to calculate the inverse mapping of original to target pixel
-    coordinates. This is a polynomial geometrical transformation.
-    Then optionally it uses a bicubic convolution interpolation
-    algorithm to map the original to target pixel values.
```

Comment:
I feel like this little bit of info should be captured *somewhere*. It's not just that our implementation of `rotate` is different from `rot` it's that it's actually using a completely different algorithm?

Reply:
I don't think it should be in the docstring of this function.  If anywhere, it should be on the known-issues page as a difference from IDL.

Regardless, I'm not versed enough in the algorithms being used to judge whether this written description of IDL's ``rot`` function is even accurate, or whether it is in fact true that there is no way to match it with any of the `scipy` or `scikit-image` interpolation orders.

Reply:
This was my understanding at the point @SolarDrew and I went all the way down this particular :rabbit: :hole:, I can no longer vouch for it's veracity.

I wouldn't say it's a known issue, I would say using an affine transform vs a warp is more appropriate to the WCS papers but I am not that fussed if it's easier to get rid of it then go for it.